pygfx/utils/bounds.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+
+
+class Bounds:
+    """Represent bounds using an axis-aligned bounding box (aabb) plus a radius.
+
+    The radius represents a sphere centered at the aabb. Semantically,
+    the bounds are the intersection of the box and the spere. This
+    allows applying transforms without growing the bounds because of
+    the box's corners.
+
+    When the radius is None, or near-equal to half the diagonal of the aabb,
+    the bounds respresents a box. The radius is at least the
+    max(width,height,depth) of the aabb, i.e. only cutting off its corners.
+    """
+
+    def __init__(self, aabb, radius=None):
```

Comment:
This seems like a good candidate for `__slots__` as well?

Reply:
Makes sense!